scripts/extract_variables_from_integration_tests.py

```diff
@@ -0,0 +1,92 @@
+"""
+extracts manageiq integration test provider data/credentials
+into variables files for the wrapanapi testsuite
+
+this script uses manageiq integration test domain knowledge
+and should be executed using its virtualenv
+"""
+
+import click
+import six
+import pathlib2
+import json
+
+
+WANTED_TAGS = frozenset(['rhevm', 'ms_scvmm'])
+
+
+def extract_rhevm_system(mgmt, credentials):
+    system_credentials = credentials[mgmt.credentials]
+    system = dict(
+        hostname=mgmt.hostname,
+        version=mgmt.version,
+        username=system_credentials.username,
+        password=system_credentials.password,
+    )
+    return dict(
+        system=system,
+        templates=mgmt.templates
+    )
+
+
+def extract_scvmm_system(mgmt, credentials):
+    system_credentials = credentials[mgmt.credentials]
+    system = dict(
+        hostname=mgmt.hostname,
+        username=system_credentials.username,
+        password=system_credentials.password,
+        domain=system_credentials.domain,
+        provisioning=mgmt.provisioning
+
+    )
+    return dict(
+        system=system,
+        templates=mgmt.templates,
+    )
+
+
+system_type_map = {
+    'scvmm': extract_scvmm_system,
+    'rhevm': extract_rhevm_system,
+}
+
+
+def extract_system(mgt, credentials):
+    return {mgt.type: system_type_map[mgt.type](mgt, credentials)}
+
+
+def write_variables(store_variables_folder, provider, data):
+    filename = "{provider}.vars.json".format(provider=provider)
+    with store_variables_folder.joinpath(filename).open('wb') as fp:
+        json.dump(data, fp, indent=2, sort_keys=True)
+
+
+def filter_mgmt_systems(management_systems):
+    for k, v in six.iteritems(management_systems):
+        if WANTED_TAGS.intersection(v.get('tags', [])):
+            yield k, v
+
+
+@click.command()
+@click.argument('store_variables_folder',
+                type=click.Path(file_okay=False, exists=False),
+                default="../wrapanapi_test_variables")
+def main(store_variables_folder):
+    store_variables_folder = pathlib2.Path(store_variables_folder)
+    if not store_variables_folder.is_dir():
+        store_variables_folder.mkdir()
+    try:
```

Comment:
**QUESTIONS:** Why not import this above like normal.   Also why the try-except, wouldn't it fail anyway?   Wait, I get the idea of having a specific error message  but would you want to do this if you had 10 local libraries to load?  Also the four libraries above should be coming from our virtualenv also.   Shouldn't they be wrapped this way if they should be wrapped this way?

Reply:
this is about fetching the yaml data from a cfme venv, and telling the user what to do if its missing